plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/AbstractPinotIntegrationSmokeTest.java

```diff
@@ -1508,6 +1560,9 @@ public void testAggregationPushdown()
         // Distinct on long is pushed down
         assertThat(query("SELECT DISTINCT long_col FROM " + ALL_TYPES_TABLE))
                 .isFullyPushedDown();
+        // Distinct on timestamp is pushed down
+        assertThat(query("SELECT DISTINCT timestamp_col FROM " + ALL_TYPES_TABLE))
```

Comment:
Does this return correct results if both a `null` and `1970-01-01T00:00:00` exist?

Reply:
the default null value for timestamps is epoch so this should return `1970-01-01T00:00:00`

Reply:
Then I would argue it's unsafe to push-down aggregates in presence of `null`s. Maybe we can instead use query-passthrough (by implementing a polymorphic table function as in https://github.com/trinodb/trino/pull/12325) for such cases to get Pinot semantics.

cc: @findepi Any opinions? i.e. DISTINCT on a column with `null` and `epoch` would return `epoch` since the `null` becomes `epoch`.

Reply:
I'm going to double check and get back to you. I think this is the same behaviour of other types like long and double

Reply:
yes, Pinot schema will assign the default null value. So this behavior is expected.

Reply:
We shouldn't push down such queries then probably. While the results match semantics of Pinot they don't match Trino so anyone performing a JOIN or SELECT across two catalogs will see inconsistent results.

For preserving Pinot semantics we should implement something like https://trino.io/docs/current/connector/postgresql.html#query-varchar-table instead.

Reply:
in pinot the only way to make distinction between null and non-null values is by using `IS NULL` and `IS NOT NULL` other than this pinot will assign the default value if null. Push-down already happens for other types, should this be addressed in a separate ticket? I could take a look at this once we merge this PR. I'll summon @elonazoulay on this one

Reply:
I tried this for filter pushdown - the behavior is not consistent with trino sql: you can select ... where is null and the result will be non null (i.e. the default value). Will take a look and see what can be done.

Reply:
yes, the projection in Pinot will always return the default value if column is null